src/main/patterns/singletonPattern/Singleton.java

```diff
@@ -0,0 +1,17 @@
+package main.patterns.singletonPattern;
+
+public final class Singleton {
+    private static Singleton object;
+    public String value;
+
+    private Singleton(String value) {
+        this.value = value;
+    }
+
+    public static Singleton getObject(String value) {
+        if (object == null) {
```

Comment:
https://refactoring.guru/ good site, except one thing.
how do you think this code will behave in multithreading environment.?

Reply:
fixed it